app/controllers/characters_controller.rb

```diff
@@ -1,15 +1,57 @@
 class CharactersController < ApplicationController
+  before_action :require_login, only: [:new, :edit, :update]
+
   def index
     @characters = Character.all
   end
 
+  def new
+    @character = Character.new
+  end
+
+  def create
+    @character = Character.new(character_params)
+    if @character.save
+      flash[:notice] = "Character created!"
+      redirect_to games_path
+    else
+      flash[:notice] = @character.errors.full_messages.join(". ")
+      render 'new'
+    end
+  end
+
   def show
     @character = Character.find(params[:id])
   end
 
+  def update
+    @character = Character.new(params[:id])
+    if @character.update(character_params)
+      flash[:notice] = "Character Added!"
+    else
+      flash[:notice] = @character.errors.full_messages.join(" ")
+    end
+  end
+
   protected
 
   def character_params
-    params.require(:character).permit(:name, :user_id)
+    params.require(:character).permit(
+                                      :name,
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Indent the first parameter one step more than the previous line.